diff.go

```diff
@@ -380,33 +391,95 @@ func diffComponentExpose(oldDef, newDef ComponentDefinition) []DiffInfo {
 
 	if oldExpose != newExpose {
 		diffInfos = append(diffInfos, DiffInfo{
-			Type: DiffInfoComponentExposeUpdated,
-			Old:  oldExpose,
-			New:  newExpose,
+			Type:      DiffTypeComponentExposeUpdated,
+			Component: componentName,
+			Old:       oldExpose,
+			New:       newExpose,
 		})
 	}
 
 	return diffInfos
 }
 
-func diffComponentScale(oldDef, newDef ComponentDefinition) []DiffInfo {
+// diffComponentScale checks in detail what diff type should be applied between
+// oldDef and newDef. The following can be applied.
+//   placement changed -> DiffTypeComponentScalePlacementUpdated
+//   min decreased     -> DiffTypeComponentScaleMinDecreased
+//   min increased     -> DiffTypeComponentScaleMinIncreased
+//   max decreased     -> DiffTypeComponentScaleMaxDecreased
+//   max increased     -> DiffTypeComponentScaleMaxIncreased
+func diffComponentScale(oldDef, newDef ComponentDefinition, componentName ComponentName) []DiffInfo {
+	oldScaleDef := &ScaleDefinition{}
```

Comment:
This does not take into account default values. This has already been noted in app-service, so we'll deal with it later.